core/templates/dev/head/components/RuleEditorDirective.js

```diff
@@ -180,6 +181,12 @@ oppia.directive('ruleEditor', ['$log', function($log) {
 
             tmpRuleDescription = tmpRuleDescription.replace(PATTERN, ' ');
           }
+
+          for (var key in $scope.rule.inputs) {
+            if (key in oldRuleInputs) {
+              $scope.rule.inputs[key] = oldRuleInputs[key];
```

Comment:
Would we need to do a type check? See the code_evaluation.py rules -- there, x could be either a CodeString or a UnicodeString.